internal/jimm/model.go

```diff
@@ -1347,3 +1351,76 @@ func (j *JIMM) ChangeModelCredential(ctx context.Context, user *openfga.User, mo
 
 	return nil
 }
+
+// ListModels list the models that the user has access to. It intentionally excludes the
+// controller model as this call is used within the context of login and register commands.
+func (j *JIMM) ListModels(ctx context.Context, user *openfga.User) ([]base.UserModel, error) {
+	const op = errors.Op("jimm.ListModels")
+	zapctx.Info(ctx, string(op))
+
+	// Get uuids of models the user has access to
+	uuids, err := user.ListModels(ctx, ofganames.ReaderRelation)
+	if err != nil {
+		return nil, errors.E(op, err, "failed to list user models")
+	}
+
+	// Get the models from the database
+	models, err := j.DB().GetModelsByUUID(ctx, uuids)
+	if err != nil {
+		return nil, errors.E(op, err, "failed to get models by uuid")
+	}
+
+	// Find the controllers these models reside on and remove duplicates
+	var controllers []dbmodel.Controller
+	seen := make(map[uint]bool)
+	for _, model := range models {
```

Comment:
since you are already iterating over models, why don't you build a map with.
map<uuid>{model}?

So later down instead of doing: 
```
if slices.Contains(uuids, um.UUID) {
  // Find that model in the db models
  index := slices.IndexFunc(models, func(m dbmodel.Model) bool {
	  return m.UUID.String == um.UUID
  })
```
you can do
if `modelmap[uuid]`.
Which is O(1) instead of O(n^2)

Reply:
nit, obviously

Reply:
Agreed, much nicer, changed